src/webob/acceptparse.py

```diff
@@ -268,6 +678,1197 @@ def _match(self, mask, item):
+                          header and `default_range`, and `default_tag` is
+                          ``None`` or not acceptable because it matches a
+                          ``q=0`` range in the header, then Lookup will next
+                          examine the `default` argument.
+
+                        | If `default` is a callable, it will be called, and
+                          the callable's return value will be returned.
+
+                        | If `default` is not a callable, the value itself will
+                          be returned.
+
+                        | The difference between supplying a ``str`` to
+                          `default_tag` and `default` is that `default_tag` is
+                          checked against ``q=0`` ranges in the header to see
+                          if it matches one of the ranges specified as not
+                          acceptable, whereas a ``str`` for the `default`
+                          argument is simply returned.
+
+                        | This parameter corresponds to the "defaulting
+                          behavior" described in :rfc:`RFC 4647, section 3.4.1
+                          <4647#section-3.4.1>`
+
+        :return: (``str``, ``None``, or any type)
+
+                 | The best match according to the Lookup matching scheme, or a
+                   return value from one of the default arguments.
+
+        **Notes**:
+
+        .. _acceptparse-lookup-asterisk-note:
+
+        - Lookup's behaviour with '*' language ranges in the header may be
+          surprising. From :rfc:`RFC 4647, section 3.4 <4647#section-3.4>`:
+
+              In the lookup scheme, this range does not convey enough
+              information by itself to determine which language tag is most
+              appropriate, since it matches everything.  If the language range
+              "*" is followed by other language ranges, it is skipped.  If the
+              language range "*" is the only one in the language priority list
+              or if no other language range follows, the default value is
+              computed and returned.
+
+          So
+
+          ::
+
+              >>> header = AcceptLanguageValidHeader('de, zh, *')
+              >>> header.lookup(language_tags=['ja', 'en'], default='default')
+              'default'
+
+        - Any tags in `language_tags` and `default_tag` and any tag matched
+          during the subtag truncation search for `default_range`, that are an
+          exact match for a non-``*`` range with ``q=0`` in the header, are
+          considered not acceptable and ruled out.
+
+        - If there is a ``*;q=0`` in the header, then `default_range` and
+          `default_tag` have no effect, as ``*;q=0`` means that all languages
+          not already matched by other ranges within the header are
+          unacceptable.
+        """
+        assert not (default_tag is None and default is None), \
```

Comment:
Instead of using assert, use an if statement. assert statements may be turned off globally, and while I tend to believe that if you break it you've bought it, I'd prefer not to have asserts in anything put testing code.

Unless it is a condition that is not likely to ever happen, and is just a sanity check.